File: apps/api/src/db/models/edge.py
# ═══════════════════════════════════════════
# ORM 模型 — 连接线表
# ═══════════════════════════════════════════

import uuid
from datetime import datetime
from sqlalchemy import String, Float, DateTime, JSON
from sqlalchemy.orm import Mapped, mapped_column
from src.db.models import Base


class EdgeModel(Base):
    __tablename__ = "edges"

    edge_id: Mapped[str] = mapped_column(String(64), primary_key=True, default=lambda: str(uuid.uuid4()))
    project_id: Mapped[str] = mapped_column(String(64), index=True, nullable=False)
    source_node_id: Mapped[str] = mapped_column(String(64), nullable=False)
    target_node_id: Mapped[str] = mapped_column(String(64), nullable=False)
    source_handle: Mapped[str | None] = mapped_column(String(64), nullable=True)
    target_handle: Mapped[str | None] = mapped_column(String(64), nullable=True)
    edge_type: Mapped[str] = mapped_column(String(32), default="strong_dependency")
    metadata_: Mapped[dict | None] = mapped_column("metadata", JSON, nullable=True)
    created_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow)

File: packages/shared-types/src/graph/edge.ts
// ═══════════════════════════════════════════
// Edge — 图结构连接线
// ═══════════════════════════════════════════

import type { EdgeType } from '../constants';

/** 连接线元数据 */
export interface EdgeMetadata {
  /** 引用强度 (0-1)，用于弱引用的权重 */
  strength?: number;
  /** 风格继承权重 (0-1) */
  inherit_weight?: number;
  /** 连接线标签（可选） */
  label?: string;
}

/** 连接线 */
export interface Edge {
  edge_id: string;
  project_id: string;
  source_node_id: string;
  target_node_id: string;
  /** 源节点的输出端口 ID */
  source_handle?: string;
  /** 目标节点的输入端口 ID */
  target_handle?: string;
  edge_type: EdgeType;
  metadata?: EdgeMetadata;
  created_at: string;
}
